aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,623 @@
+/// A work in progress NFT DAO Platform Example
+/// An example e2e flow.
+/// There are multiple roles: DAO platform operator, DAO creator, proposer and voter.
+/// 1. platform operator deploy this package to create a DAO platform
+/// 2. DAO creator calls `create_dao` to create their DAO.
+/// 3. A proposer can specify the DAO she wants to create a proposal and create the proposal through  `create_proposal`
+///     two types of proposals can be created:
+///     3.a: no-op, no automatic execution happens on chain. Only the proposal and its results are recorded on-chain for DAO admin to take actions
+///     3.b: transfer fund, the fund will be automatically transferred from DAO resource account to the destination account.
+/// 4. A voter can vote for a proposal of a DAO through `vote`
+/// 5. Anyone can call the `resolve` to resolve a proposal. A proposal voting duration has to expire and the proposal should have more votes than the minimal required threshold.
+///
+/// An example of DAO delegation flow
+///  1. the DAO creator can delegate her DAO through `delegate_dao` to another admin address
+///
+/// An example of DAO removal.
+/// 1. The DAO creator can call `reclaim_signer_capability` to remove her DAO from the platform and get back her resource account's signercapability
+module dao_platform::nft_dao {
+    use aptos_framework::account::{SignerCapability, create_signer_with_capability, get_signer_capability_address};
+    use aptos_framework::account;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::coin;
+    use aptos_framework::timestamp;
+    use aptos_std::bucket_table::BucketTable;
+    use aptos_std::bucket_table;
+    use aptos_std::table::Table;
+    use aptos_std::table;
+    use aptos_token::property_map::PropertyMap;
+    use aptos_token::property_map;
+    use aptos_token::token::{Self, TokenId};
+    use dao_platform::nft_dao_events::emit_create_dao_event;
+    use dao_platform::token_type::get_token_type;
+    use std::bcs;
+    use std::error;
+    use std::signer;
+    use std::string::String;
+    use std::string;
+    use std::vector;
+
+    /// The new address to be delegated to already exists in the platform
+    const EDELEGATED_ADDRESS_ALREADY_EXISTS: u64 = 1;
+
+    /// This account doesn't have enough voting weights
+    const EVOTING_WEIGHTS_NOT_ENOUGH: u64 = 2;
+
+    /// This account doesn't own this DAO's voting token
+    const ENOT_OWN_THE_VOTING_DAO_TOKEN: u64 = 3;
+
+    /// This function is supported in proposal
+    const ENOT_SUPPROTED_FUNCTION: u64 = 4;
+
+    /// Can only propose a start time in future
+    const EPROPOSAL_START_TIME_SHOULD_IN_FUTURE: u64 = 5;
+
+    /// Cannot only use membership token of the DAO
```

Comment:
This error message is a bit hard to understand. Can you make it clearer? 🙏

Reply:
remove this one